pkg/query-service/model/errors.go

```diff
@@ -0,0 +1,36 @@
+package model
+
+import "fmt"
+
+// custom errors related to registration
+type ErrFeatureUnavailable struct {
+	Key string
+}
+
+func (errFeatureUnavailable ErrFeatureUnavailable) Error() string {
+	return fmt.Sprintf("feature unavailable: %s", errFeatureUnavailable.Key)
+}
+
+type ErrEmailRequired struct{}
+
+func (errEmailRequired ErrEmailRequired) Error() string {
+	return "email is required"
+}
```

Comment:
I understand the intent behind this but there is no data associated with these error types. What do you think about defining them with `errors.New`/`fmt.Errorf`?

Reply:
I think we should adopt named error codes as a pattern. This will help with a centralised control on error messages. especially if we consider applying  translations (internationalisation) in future.  

Most well-known libraries take a named errors approach as well. e.g. AWS sdk

The error structs can evolve contain into internal message (to be written to log) for developers and the error message to be displayed to the end user.  At the moment, we just show 'something went wrong'.  Printing internal error on web console (developer console) will help with additional context.   Having requestId will also help.

In this PR, I could not make all changes at once but wanted to initiate this change atleast for generic and common errors.

Reply:
I am going to skip this for now. We can discuss more, as I will log a separate issue for named errors. I think if we are looking translations (internationalisation) a struct enables adding language code. Usually language code will come from the current user.   

The front-end already takes care of majority of translations but doesn't have control over the error messages. 

e.g. 
ErrEmailRequired {
Lang: string;
} 

func (e *ErrEmailRequired) Error() string {
case e.Lang {
case English.US: 
 return 'Email is required''
Case Spanish:
.... 
}